### ISSUE TITLE
Implement from scratch a KDE-based predictor

### ISSUE THREAD

**Issue by chengsoonong** [chengsoonong/acton#2]
Implement a new predictor in the sklearn style that works as follows:
1. Fit each class with [KDE](http://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KernelDensity.html#sklearn.neighbors.KernelDensity) with Gaussian kernel, Euclidean metric
2. Use score_samples on each item of the test set
3. Return the most probable class

A bit of fiddling would need to be done to find a good bandwidth, suggest median value of squared Euclidean distance of all training points.
